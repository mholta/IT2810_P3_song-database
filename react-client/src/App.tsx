import React from 'react';
import MainRouter from './pages/MainRouter';
import { BrowserRouter as Router } from 'react-router-dom';
import { ThemeProvider } from '@mui/material';
import { theme } from './styles/theme';
import { library } from '@fortawesome/fontawesome-svg-core';
import { fab } from '@fortawesome/free-brands-svg-icons';
import {
  faHome,
  faMousePointer,
  faMusic,
} from '@fortawesome/free-solid-svg-icons';
import { Provider as StoreProvider } from 'react-redux';
import store from './store/store';
import PageLayoutWrapper from './components/layout/PageLayoutWrapper';

/* Add icons from Font Awesome */
library.add(fab, faMousePointer, faHome, faMusic);

const App = () => {
  return (
    <StoreProvider store={store}>
      <ThemeProvider theme={theme}>
        <Router>
          <PageLayoutWrapper>
            <MainRouter />
          </PageLayoutWrapper>
        </Router>
      </ThemeProvider>
    </StoreProvider>
  );
};

export default App;
