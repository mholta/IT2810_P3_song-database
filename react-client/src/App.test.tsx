import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

test('renders successfully', () => {
  const tree = renderer.create(<App />).toJSON();

  expect(tree).toMatchSnapshot();
});
