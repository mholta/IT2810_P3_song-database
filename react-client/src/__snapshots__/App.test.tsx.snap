// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders successfully 1`] = `
<div
  className="css-185wnrw"
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <nav
    className="css-1ebp07k"
    open={false}
  >
    <div
      className="css-ydsbw5"
    >
      <a
        className="css-1yg7sek"
        href="/"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-home fa-w-18 "
          data-icon="home"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         
        <span>
          Hjem
        </span>
      </a>
      <a
        className="css-1yg7sek"
        href="/search"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-music fa-w-16 "
          data-icon="music"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M470.38 1.51L150.41 96A32 32 0 0 0 128 126.51v261.41A139 139 0 0 0 96 384c-53 0-96 28.66-96 64s43 64 96 64 96-28.66 96-64V214.32l256-75v184.61a138.4 138.4 0 0 0-32-3.93c-53 0-96 28.66-96 64s43 64 96 64 96-28.65 96-64V32a32 32 0 0 0-41.62-30.49z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         
        <span>
          Sanger
        </span>
      </a>
      <a
        className="css-1yg7sek"
        href="/search?query=søker på dette"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-music fa-w-16 "
          data-icon="music"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M470.38 1.51L150.41 96A32 32 0 0 0 128 126.51v261.41A139 139 0 0 0 96 384c-53 0-96 28.66-96 64s43 64 96 64 96-28.66 96-64V214.32l256-75v184.61a138.4 138.4 0 0 0-32-3.93c-53 0-96 28.66-96 64s43 64 96 64 96-28.65 96-64V32a32 32 0 0 0-41.62-30.49z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         
        <span>
          Søk på noe
        </span>
      </a>
      <a
        className="css-1yg7sek"
        href="/song/testsang"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-music fa-w-16 "
          data-icon="music"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M470.38 1.51L150.41 96A32 32 0 0 0 128 126.51v261.41A139 139 0 0 0 96 384c-53 0-96 28.66-96 64s43 64 96 64 96-28.66 96-64V214.32l256-75v184.61a138.4 138.4 0 0 0-32-3.93c-53 0-96 28.66-96 64s43 64 96 64 96-28.65 96-64V32a32 32 0 0 0-41.62-30.49z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         
        <span>
          Se test-sang
        </span>
      </a>
    </div>
  </nav>
  <div
    className="css-9c3fle"
  >
    <div
      className="css-cssveg"
    >
      <div
        className="css-0"
      >
        <form
          action="/search"
          className="css-ofpocj"
          onChange={[Function]}
          onSubmit={[Function]}
        >
          <div
            className="css-1rmzhqg"
          >
            <div
              className="css-1f7apd6"
            >
              <div
                className="css-18a5p6x"
                focused={0}
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="SearchIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </div>
              <input
                autoComplete="off"
                className="css-psh2zg"
                name="query"
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                open={0}
                placeholder="Søk på sang"
                type="search"
                value=""
              />
            </div>
          </div>
        </form>
      </div>
    </div>
    <main
      className="css-1tbz3yy"
    >
      <div
        className="css-2x3ewd"
      >
        <div>
          HomePage
        </div>
      </div>
    </main>
  </div>
</div>
`;
